Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java

```diff
@@ -11,92 +11,91 @@
-			}
-			else
-			{
-				haveAccess = authorization.chkAuthorizationByRole(userName, policy.userRole());
-			}
-			
-			if(haveAccess){
-				break;
-			}
-		}
-		
-		if(!haveAccess)
-		{
-			return restAccessException.accessErrorMsg(400,"You don't have permission");
-		}
-		Object retVal = pjp.proceed();
-		return retVal;
-	}
-	
+public class RestAccessAspect {
+
+    @Autowired
+    private IAuthorizationManager authorizationManager;
+
+    /**
+     * This method prevents the access permissions check for the web package
+     * methods annotated with 'noCheck' for rest interfaces
+     * 
+     * @param pjp
+     * @param noCheck
+     * @return ProceedingJoinPoint object
+     * @throws Throwable
+     */
+    @Around("within(edu.asu.spring.quadriga.rest..*) && @annotation(noCheck)")
+    public Object chkProjectAuthorization(ProceedingJoinPoint pjp, NoAuthorizationCheck noCheck) throws Throwable {
+        return pjp.proceed();
+    }
+
+    /**
+     * This method checks the access permissions for the objects before
+     * executing the methods.This provides run time access permission check to
+     * the methods in the web package and annotated with 'checks' during rest
+     * interface access.
+     * 
+     * @param pjp
+     * @param checks
+     * @return - no access it throws Access Denied exception. if he have access
+     *         returns ProceedingJointPoint object
+     * @throws Throwable
+     */
+    @Around("within(edu.asu.spring.quadriga.rest..*) && @annotation(checks)")
+    public Object chkAuthorization(ProceedingJoinPoint pjp, RestAccessPolicies checks) throws Throwable {
+        boolean haveAccess;
+        String userName;
+        String accessObjectId;
+        IAuthorization authorization;
+
+        haveAccess = true;
+        try {
+            // retrieve the logged in User name
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            userName = auth.getName();
+            // Loop through all the access policies specified
+            ElementAccessPolicy[] policies = checks.value();
+
+            for (ElementAccessPolicy policy : policies) {
+                // retrieve the authorization object based on the type
+                authorization = authorizationManager.getAuthorizationObject(policy.type());
+
+                // calling the object
+                if (policy.paramIndex() > 0) {
+                    accessObjectId = pjp.getArgs()[policy.paramIndex() - 1].toString();
+
+                    haveAccess = authorization.chkAuthorization(userName, accessObjectId, policy.userRole());
+
+                } else {
+                    haveAccess = authorization.chkAuthorizationByRole(userName, policy.userRole());
+                }
+
+                if (haveAccess) {
+                    break;
+                }
+            }
+        } catch (QuadrigaAccessException e) {

```

Comment:
this catch block should be where the exception is actually thrown (line 80?), not for the whole block.